sql/plan/show_create_procedure.go

```diff
@@ -0,0 +1,128 @@
+	&sql.Column{Name: "sql_mode", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "Create Procedure", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "character_set_client", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "collation_connection", Type: sql.LongText, Nullable: false},
+	&sql.Column{Name: "Database Collation", Type: sql.LongText, Nullable: false},
+}
+
+// NewShowCreateProcedure creates a new ShowCreateProcedure node for SHOW CREATE PROCEDURE statements.
+func NewShowCreateProcedure(db sql.Database, procedure string) *ShowCreateProcedure {
+	return &ShowCreateProcedure{
+		db:            db,
+		ProcedureName: strings.ToLower(procedure),
+	}
+}
+
+// String implements the sql.Node interface.
+func (s *ShowCreateProcedure) String() string {
+	return fmt.Sprintf("SHOW CREATE PROCEDURE %s", s.ProcedureName)
+}
+
+// Resolved implements the sql.Node interface.
+func (s *ShowCreateProcedure) Resolved() bool {
+	_, ok := s.db.(sql.UnresolvedDatabase)
+	return !ok
+}
+
+// Children implements the sql.Node interface.
+func (s *ShowCreateProcedure) Children() []sql.Node {
+	return nil
+}
+
+// Schema implements the sql.Node interface.
+func (s *ShowCreateProcedure) Schema() sql.Schema {
+	return showCreateProcedureSchema
+}
+
+// RowIter implements the sql.Node interface.
+func (s *ShowCreateProcedure) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	procedureDb, ok := s.db.(sql.StoredProcedureDatabase)
+	if !ok {
+		return nil, sql.ErrStoredProceduresNotSupported.New(s.db.Name())
+	}
+	procedures, err := procedureDb.GetStoredProcedures(ctx)
+	if err != nil {
+		return nil, err
+	}
+	for _, procedure := range procedures {
+		if strings.ToLower(procedure.Name) == s.ProcedureName {
+			characterSetClient, err := ctx.GetSessionVariable(ctx, "character_set_client")
+			if err != nil {
+				return nil, err
+			}
+			collationConnection, err := ctx.GetSessionVariable(ctx, "collation_connection")
+			if err != nil {
+				return nil, err
+			}
+			collationServer, err := ctx.GetSessionVariable(ctx, "collation_server")
+			if err != nil {
+				return nil, err
+			}
+			return sql.RowsToRowIter(sql.Row{
+				procedure.Name,            // Procedure
+				"",                        // sql_mode
+				procedure.CreateStatement, // Create Procedure
+				characterSetClient,        // character_set_client
+				collationConnection,       // collation_connection
+				collationServer,           // Database Collation
+			}), nil
+		}
+	}
+	return nil, sql.ErrStoredProcedureDoesNotExist.New(s.ProcedureName)
+}
+
+// WithChildren implements the sql.Node interface.
+func (s *ShowCreateProcedure) WithChildren(children ...sql.Node) (sql.Node, error) {
+	return NillaryWithChildren(s, children...)
+}
+
+// CheckPrivileges implements the interface sql.Node.
+func (s *ShowCreateProcedure) CheckPrivileges(ctx *sql.Context, opChecker sql.PrivilegedOperationChecker) bool {
+	//TODO: figure out what privileges are needed here
```

Comment:
Privileges used are the 2nd paragraph:
https://dev.mysql.com/doc/refman/8.0/en/show-create-procedure.html
I’d keep a TODO for the definer since it’s not being set yet